lib/parser.js

```diff
@@ -202,15 +207,39 @@ async function customDocumentOperations(parsedJSON, asyncapiYAMLorJSON, initialF
   validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat);
   validateOperationId(parsedJSON, asyncapiYAMLorJSON, initialFormat, OPERATIONS);
 
-  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
-  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
+  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options, schemaValidationErrors);
+  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options, schemaValidationErrors);
 }
 
-async function validateAndConvertMessage(msg, originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload) {
+async function validateAndConvertMessage(msg,name ,originalAsyncAPIDocument, fileFormat, parsedAsyncAPIDocument, pathToPayload, options, schemaValidationErrors) {
   //check if the message has been parsed before
   if (xParserMessageParsed in msg && msg[String(xParserMessageParsed)] === true) return;
   const defaultSchemaFormat = getDefaultSchemaFormat(parsedAsyncAPIDocument.asyncapi);
   const schemaFormat = msg.schemaFormat || defaultSchemaFormat;
+  const parser = PARSERS[String(schemaFormat)];
+  const errorMap = new Map();
+  if(options.component){
+    errorMap.set(`messages/${name}/${schemaFormat}`, schemaFormat);  
+  }
+  else{
+    errorMap.set(`${tilde(options.channelName)}/${name}/${msg}/${schemaFormat}`, schemaFormat);
+  }
+  
+  if (!parser) {
+    schemaValidationErrors.push(new ParserError({
+      type: 'schema-parser-not-registered',
+      title: 'The specification has an unsupported schema format',
+      detail:
+        'You shall not pass, no schema parser available that can parse this schema. The good news is that we have one for Avro, just install @asyncapi/avro-schema-parser and use registerSchemaParser to register it with the current parser',
+      validationErrors: groupValidationErrors(
+        'channels',
+        'channel does not have a corresponding schema object',
+        errorMap,
+        originalAsyncAPIDocument,
+        parser
+      ),
+    }));
+  }
 
   await PARSERS[String(schemaFormat)]({
```

Comment:
this is the reason you tests fail, because even though you are catching errors above, you still go and execute parser that might not exist. This await and also rest of the function should go into `else` so it is called only if `parser` is really there.

also you can call `customSchemaParser` that you have above and not `PARSERS[String(schemaFormat)]` again